.github/workflows/e2e.yml

```diff
@@ -1,85 +1,209 @@
+# This workflow file contains end-to-end tests for running client apps on
+# our baremetal testbed. Artifacts are intentionally built in debug mode
+# so that debug assert and other test-related macros can be triggered.
+#
+# This workflow does not replace the need to sanity check the QA builds manually,
+# but should reduce (and some day) eliminate the need for a full manual QA runbook.
 name: End to end tests
 on:
   # TODO
   # push:
   #   branches:
   #     - main
   workflow_dispatch:
-
-defaults:
-  working-directory: ./e2e
+  pull_request:
 
 concurrency:
   group: "e2e-${{ github.workflow }}-${{ github.ref }}"
   cancel-in-progress: true
 
 jobs:
-  setup-e2e:
+  setup:
     runs-on: ubuntu-22.04-firezone
+    env:
+      # mark:automatic-version
+      VERSION: "1.0.0"
     steps:
       - uses: actions/checkout@v4
-      - run: echo "Spin up docker test environment"
+      - run: docker compose up -d postgres
+      - run: docker compose run elixir /bin/sh -c 'cd apps/domain && mix do ecto.create, ecto.migrate, ecto.seed'
+      - name: Build images
+        run: |
+          docker compose build
+      - name: Start docker compose in the background
+        run: |
+          docker compose up -d \
+            api \
+            web \
+            client \
+            relay \
+            gateway \
+            iperf3
+      - name: Wait for client to ping resource
+        run: |
+          docker compose exec -it client timeout 60 \
+          sh -c 'until ping -W 1 -c 1 172.20.0.100 &>/dev/null; do true; done'
 
   macos:
-    needs: setup-e2e
+    needs: setup
     runs-on: macos-14-firezone
+    permissions:
+      contents: read
+      id-token: 'write'
+    defaults:
+      run:
+        working-directory: ./swift/apple
     steps:
       - uses: actions/checkout@v4
-      - run: echo 'Noop'
+      - uses: ./.github/actions/setup-rust
+        with:
+          sccache_enabled: false
+          targets: aarch64-apple-darwin
+      - name: build macos app
+        run: |
+          # Copy xcconfig
+          cp Firezone/xcconfig/debug.xcconfig Firezone/xcconfig/config.xcconfig
 
-  windows:
-    needs: setup-e2e
-    runs-on: windows-11-firezone
-    steps:
-      - uses: actions/checkout@v4
-      - run: echo 'Noop'
+          # Unlock keychain; simulates a user logged into the GUI
+          security unlock-keychain -p '${{ secrets.MACOS_SELF_HOSTED_USER_PASSWORD }}'
```

Comment:
"MACOS"?

Reply:
Yeah this is the local user account password for the `firezone` user as stored in GH secrets.